Document the release flag support (#13668)
gradle/gradle: subprojects/docs/src/docs/userguide/jvm/building_java_projects.adoc

That's also how you can change the verbosity of the compiler, disable debug output in the byte code and configure where the compiler can find annotation processors.

[[sec:java_cross_compilation]]
=== Targeting a specific Java version

By default, Gradle will compile Java code to the language level of the JVM running Gradle.

> Since version 9, the Java compiler can be configured to produce bytecode for an older Java version, additionally making sure the code does not use any APIs from a more recent version.

Reviewer: ```suggestion
Since version 9, the Java compiler can be configured to produce bytecode for an older Java version while making sure the code does not use any APIs from a more recent version.
```

Gradle now supports this link:{groovyDslPath}/org.gradle.api.tasks.compile.CompileOptions.html#org.gradle.api.tasks.compile.CompileOptions:release[release] flag on `CompileOptions` directly for Java compilation.
This option takes precedence over the properties described below.

> Historical options for the Java compiler remain available, at the project level, on the `java` extension:

Reviewer: You can also set them on the compile task directly. Maybe just shorten the sentence?

```suggestion
Historical options for the Java compiler remain available:
```

Okay now I see that this is mentioned below. :) That's a bit weird order. So maybe keep it shorter here and explain _all_ details below 👇 .

`sourceCompatibility`::
Defines which language version of Java your source files should be treated as.

`targetCompatibility`::
Defines the minimum JVM version your code should run on, i.e. it determines the version of byte code the compiler generates.

> Alternatively, these options can be set per link:{groovyDslPath}/org.gradle.api.tasks.compile.JavaCompile.html[JavaCompile] task, using properties with the same names.

Reviewer: ```suggestion
These options can be set per link:{groovyDslPath}/org.gradle.api.tasks.compile.JavaCompile.html[JavaCompile] task, or on the `java { }` extension for all compile tasks, using properties with the same names.
```

However these options do not protect against the use of APIs introduced in later Java versions.

==== Compiling and testing Java 6/7

Gradle can only run on Java version 8 or higher.

> [NOTE]

Reviewer: This note feels out of place here after only the sentence above. The sentence above and below could use a rewrite to clarify that this is needed if you actually want to run an older Java executable. And that you can use "release" if it is just about the version...

Or you just move this note below the next sentence. :) (Which could be combined into one paragraph with the sentence above).

Author: Moved indeed
